RuntimeException while uploading image if the license if invalid

**Summary:** 
While uploading a media, if the license is somehow invalid we throw a RuntimeException and as it is not handled, the app crashes

**Steps to reproduce:** 

How can we reproduce the issue? 
Provide invalid license while uploading image.

**System logs:**

```
java.lang.RuntimeException: Unable to resume activity {fr.free.nrw.commons/fr.free.nrw.commons.upload.UploadActivity}: java.lang.RuntimeException: Unrecognized license value: CC BY-SA
at android.app.ActivityThread.performResumeActivity(ActivityThread.java:3421)
at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:3461)
at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2730)
at android.app.ActivityThread.-wrap12(ActivityThread.java)
at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1473)
at android.os.Handler.dispatchMessage(Handler.java:102)
at android.os.Looper.loop(Looper.java:154)
at android.app.ActivityThread.main(ActivityThread.java:6123)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:867)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:757)
Caused by: java.lang.RuntimeException: Unrecognized license value: CC BY-SA
at fr.free.nrw.commons.Utils.licenseUrlFor(Utils.java:135)
at fr.free.nrw.commons.upload.UploadActivity.updateLicenseSummary(UploadActivity.java:275)
at fr.free.nrw.commons.upload.UploadPresenter.updateLicenses(UploadPresenter.java:335)
at fr.free.nrw.commons.upload.UploadPresenter.addView(UploadPresenter.java:312)
at fr.free.nrw.commons.upload.UploadActivity.lambda$checkStoragePermissions$2$UploadActivity(UploadActivity.java:204)
at fr.free.nrw.commons.upload.UploadActivity$$Lambda$4.run(Unknown Source)
at fr.free.nrw.commons.utils.PermissionUtils$1.onPermissionGranted(PermissionUtils.java:77)
at com.karumi.dexter.MultiplePermissionsListenerToPermissionListenerAdapter.onPermissionsChecked(Unknown Source)
at com.karumi.dexter.DexterInstance$1.run(Unknown Source)
at com.karumi.dexter.MainThread.execute(Unknown Source)
at com.karumi.dexter.DexterInstance.checkMultiplePermissions(Unknown Source)
at com.karumi.dexter.DexterInstance.checkPermissions(Unknown Source)
at com.karumi.dexter.Dexter.check(Unknown Source)
at fr.free.nrw.commons.utils.PermissionUtils.checkPermissionsAndPerformAction(PermissionUtils.java:103)
at fr.free.nrw.commons.upload.UploadActivity.checkStoragePermissions(UploadActivity.java:202)
at fr.free.nrw.commons.upload.UploadActivity.onResume(UploadActivity.java:190)
at android.app.Instrumentation.callActivityOnResume(Instrumentation.java:1291)
at android.app.Activity.performResume(Activity.java:6776)
at android.app.ActivityThread.performResumeActivity(ActivityThread.java:3398)
... 10 more
java.lang.RuntimeException: Unrecognized license value: CC BY-SA
at fr.free.nrw.commons.Utils.licenseUrlFor(Utils.java:135)
at fr.free.nrw.commons.upload.UploadActivity.updateLicenseSummary(UploadActivity.java:275)
at fr.free.nrw.commons.upload.UploadPresenter.updateLicenses(UploadPresenter.java:335)
at fr.free.nrw.commons.upload.UploadPresenter.addView(UploadPresenter.java:312)
at fr.free.nrw.commons.upload.UploadActivity.lambda$checkStoragePermissions$2$UploadActivity(UploadActivity.java:204)
at fr.free.nrw.commons.upload.UploadActivity$$Lambda$4.run(Unknown Source)
at fr.free.nrw.commons.utils.PermissionUtils$1.onPermissionGranted(PermissionUtils.java:77)
at com.karumi.dexter.MultiplePermissionsListenerToPermissionListenerAdapter.onPermissionsChecked(Unknown Source)
at com.karumi.dexter.DexterInstance$1.run(Unknown Source)
at com.karumi.dexter.MainThread.execute(Unknown Source)
at com.karumi.dexter.DexterInstance.checkMultiplePermissions(Unknown Source)
at com.karumi.dexter.DexterInstance.checkPermissions(Unknown Source)
at com.karumi.dexter.Dexter.check(Unknown Source)
at fr.free.nrw.commons.utils.PermissionUtils.checkPermissionsAndPerformAction(PermissionUtils.java:103)
at fr.free.nrw.commons.upload.UploadActivity.checkStoragePermissions(UploadActivity.java:202)
at fr.free.nrw.commons.upload.UploadActivity.onResume(UploadActivity.java:190)
at android.app.Instrumentation.callActivityOnResume(Instrumentation.java:1291)
at android.app.Activity.performResume(Activity.java:6776)
at android.app.ActivityThread.performResumeActivity(ActivityThread.java:3398)
at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:3461)
at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2730)
at android.app.ActivityThread.-wrap12(ActivityThread.java)
at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1473)
at android.os.Handler.dispatchMessage(Handler.java:102)
at android.os.Looper.loop(Looper.java:154)
at android.app.ActivityThread.main(ActivityThread.java:6123)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:867)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:757)
```

**Device and Android version:** 

WUSER_COMMENT=
ANDROID_VERSION=7.0
BRAND=motorola
PHONE_MODEL=Moto G (4)
**Commons app version:** 
APP_VERSION_NAME=2.10.2

**Would you like to work on the issue?**
Yes

--- a/app/src/main/java/fr/free/nrw/commons/Utils.java
+++ b/app/src/main/java/fr/free/nrw/commons/Utils.java
@@ -103,12 +103,8 @@ public class Utils {
                 return R.string.license_name_cc_by_sa_four;
             case Prefs.Licenses.CC0:
                 return R.string.license_name_cc0;
-            case Prefs.Licenses.CC_BY:  // for backward compatibility to v2.1
-                return R.string.license_name_cc_by_3_0;
-            case Prefs.Licenses.CC_BY_SA:  // for backward compatibility to v2.1
-                return R.string.license_name_cc_by_sa_3_0;
         }
-        throw new RuntimeException("Unrecognized license value: " + license);
+        throw new IllegalStateException("Unrecognized license value: " + license);
     }
 
     /**
@@ -132,7 +128,7 @@ public class Utils {
             case Prefs.Licenses.CC0:
                 return "https://creativecommons.org/publicdomain/zero/1.0/";
             default:
-                throw new RuntimeException("Unrecognized license value: " + license);
+                throw new IllegalStateException("Unrecognized license value: " + license);
         }
     }
 

--- a/app/src/main/java/fr/free/nrw/commons/contributions/Contribution.java
+++ b/app/src/main/java/fr/free/nrw/commons/contributions/Contribution.java
@@ -258,10 +258,6 @@ public class  Contribution extends Media {
                 return "{{self|cc-by-sa-4.0}}";
             case Prefs.Licenses.CC0:
                 return "{{self|cc-zero}}";
-            case Prefs.Licenses.CC_BY:
-                return "{{self|cc-by-3.0}}";
-            case Prefs.Licenses.CC_BY_SA:
-                return "{{self|cc-by-sa-3.0}}";
         }
 
         throw new RuntimeException("Unrecognized license value: " + license);

--- a/app/src/main/java/fr/free/nrw/commons/settings/Prefs.java
+++ b/app/src/main/java/fr/free/nrw/commons/settings/Prefs.java
@@ -14,9 +14,5 @@ public class Prefs {
         public static final String CC_BY_SA_4 = "CC BY-SA 4.0";
         public static final String CC_BY_4 = "CC BY 4.0";
         public static final String CC0 = "CC0";
-
-        // kept for backward compatibility to v2.1
-        @Deprecated public static final String CC_BY = "CC BY";
-        @Deprecated public static final String CC_BY_SA = "CC BY-SA";
     }
 }

--- a/app/src/main/java/fr/free/nrw/commons/upload/UploadPresenter.java
+++ b/app/src/main/java/fr/free/nrw/commons/upload/UploadPresenter.java
@@ -2,16 +2,8 @@ package fr.free.nrw.commons.upload;
 
 import android.annotation.SuppressLint;
 import android.content.Context;
-
-import java.lang.reflect.Proxy;
-import java.util.ArrayList;
-import java.util.List;
-
-import javax.inject.Inject;
-import javax.inject.Named;
-import javax.inject.Singleton;
-
 import fr.free.nrw.commons.R;
+import fr.free.nrw.commons.Utils;
 import fr.free.nrw.commons.category.CategoriesModel;
 import fr.free.nrw.commons.contributions.Contribution;
 import fr.free.nrw.commons.filepicker.UploadableFile;
@@ -23,6 +15,12 @@ import fr.free.nrw.commons.utils.StringUtils;
 import io.reactivex.Observable;
 import io.reactivex.android.schedulers.AndroidSchedulers;
 import io.reactivex.schedulers.Schedulers;
+import java.lang.reflect.Proxy;
+import java.util.ArrayList;
+import java.util.List;
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
 import timber.log.Timber;
 
 import static fr.free.nrw.commons.upload.UploadModel.UploadItem;
@@ -330,7 +328,15 @@ public class UploadPresenter {
      * Sets the list of licences and the default license.
      */
     private void updateLicenses() {
-        String selectedLicense = defaultKvStore.getString(Prefs.DEFAULT_LICENSE, Prefs.Licenses.CC_BY_SA_3);
+        String selectedLicense = defaultKvStore.getString(Prefs.DEFAULT_LICENSE,
+            Prefs.Licenses.CC_BY_SA_4);//CC_BY_SA_4 is the default one used by the commons web app
+        try {//I have to make sure that the stored default license was not one of the deprecated one's
+            Utils.licenseNameFor(selectedLicense);
+        } catch (IllegalStateException exception) {
+            Timber.e(exception.getMessage());
+            selectedLicense = Prefs.Licenses.CC_BY_SA_4;
+            defaultKvStore.putString(Prefs.DEFAULT_LICENSE, Prefs.Licenses.CC_BY_SA_4);
+        }
         view.updateLicenses(uploadModel.getLicenses(), selectedLicense);
         view.updateLicenseSummary(selectedLicense, uploadModel.getCount());
     }

